File: frontend/src/services/component.ts
import { api } from "./api";

export const ComponentService = {
  async create(
    architectureId: string,
    componentType: string
  ) {

    const payload = {
      name:
        componentType +
        "-" +
        Date.now(),

      component_type:
        componentType,

      config: {},
    };

    const response =
      await api.post(
        `/architectures/${architectureId}/components`,
        payload
      );

    return response.data;
  },
  async updatePosition(
    componentId: string,
    x: number,
    y: number
  ) {
    return api.patch(
      `/components/${componentId}/position`,
      {
        x_position: x,
        y_position: y,
      }
    );
  }
};

File: backend/app/api/v1/component.py
# app/api/v1/components.py
from uuid import UUID

from fastapi import APIRouter, Depends
from sqlmodel import Session

from app.api.dependencies import get_db
from app.schemas.component import PositionUpdate
from app.services.component_service import ComponentService

router = APIRouter(
    prefix="/components",
    tags=["Components"]
)

@router.patch("/{component_id}/position")
def update_position(
    component_id: UUID,
    payload: PositionUpdate,
    db: Session = Depends(get_db),
):
    return ComponentService.update_position(
        db=db,
        component_id=component_id,
        x=payload.x_position,
        y=payload.y_position,
    )